vm/src/builtins/union.rs

```diff
@@ -90,15 +90,13 @@ impl PyUnion {
     }
 
     #[pymethod(magic)]
-    fn instancecheck(_zelf: PyRef<Self>, _obj: PyObjectRef, vm: &VirtualMachine) -> PyResult {
-        Err(vm
-            .new_type_error("isinstance() argument 2 cannot be a parameterized generic".to_owned()))
+    fn instancecheck(zelf: PyRef<Self>, obj: PyObjectRef, vm: &VirtualMachine) -> PyResult<bool> {
+        obj.is_instance(zelf.args().as_object(), vm)
     }
 
     #[pymethod(magic)]
-    fn subclasscheck(_zelf: PyRef<Self>, _obj: PyObjectRef, vm: &VirtualMachine) -> PyResult {
-        Err(vm
-            .new_type_error("issubclass() argument 2 cannot be a parameterized generic".to_owned()))
+    fn subclasscheck(zelf: PyRef<Self>, obj: PyObjectRef, vm: &VirtualMachine) -> PyResult<bool> {
+        obj.is_subclass(zelf.args().as_object(), vm)
```

Comment:
We need to check whether `GenericAlias` object exist in args and then return `TypeError` with message "isinstance() argument 2 cannot contain a parameterized generic"

```python
>>>>> isinstance(1, int | list[str])
# rustpython
True
# cpython
TypeError: isinstance() argument 2 cannot contain a parameterized generic
```

Reply:
Hmm... As `is_instance` and `is_subclass` in the `object.rs` are already checking each argument in the tuple, so we don't have to iterate args and call `is_instance` or `is_subclass`. I also agree with your implmenetation.

But in cpython, they switched to the implementation that iterates all the args in `union` and call `is_instance` or `is_subclass` for each one. (https://github.com/python/cpython/pull/27120)

I don't know the exact reason yet, haha.. 😂

Reply:
Thx for recognizing me!